src/lang/modifyAst/boolean.ts

```diff
@@ -0,0 +1,346 @@
+import { Node } from '@rust/kcl-lib/bindings/Node'
+import CodeManager from 'lang/codeManager'
+import { KclManager } from 'lang/KclSingleton'
+import {
+  createArrayExpression,
+  createCallExpressionStdLibKw,
+  createLabeledArg,
+  createLocalName,
+  createVariableDeclaration,
+  findUniqueName,
+} from 'lang/modifyAst'
+import { getNodeFromPath } from 'lang/queryAst'
+import { getNodePathFromSourceRange } from 'lang/queryAstNodePathUtils'
+import { getFaceCodeRef } from 'lang/std/artifactGraph'
+import { EngineCommandManager } from 'lang/std/engineConnection'
+import {
+  Artifact,
+  ArtifactGraph,
+  Program,
+  VariableDeclaration,
+} from 'lang/wasm'
+import { Selection, Selections } from 'lib/selections'
+import { err } from 'lib/trap'
+import { isArray } from 'lib/utils'
+
+export async function applySubtractFromTargetOperatorSelections(
+  target: Selection,
+  tool: Selection,
+  dependencies: {
+    kclManager: KclManager
+    engineCommandManager: EngineCommandManager
+    codeManager: CodeManager
+  }
+): Promise<Error | void> {
+  const ast = dependencies.kclManager.ast
+  if (!target.artifact || !tool.artifact) {
+    return new Error('No artifact found')
+  }
+  const orderedChildrenTarget = findAllChildrenAndOrderByPlaceInCode(
+    target.artifact,
+    dependencies.engineCommandManager.artifactGraph
+  )
+  const orderedChildrenTool = findAllChildrenAndOrderByPlaceInCode(
+    tool.artifact,
+    dependencies.engineCommandManager.artifactGraph
+  )
+
+  const lastVarTarget = getLastVariable(orderedChildrenTarget, ast)
+  const lastVarTool = getLastVariable(orderedChildrenTool, ast)
+
+  if (!lastVarTarget || !lastVarTool) {
+    return new Error('No variable found')
+  }
+  const modifiedAst = booleanSubtractAstMod({
+    ast,
+    targets: [lastVarTarget?.variableDeclaration?.node],
+    tools: [lastVarTool?.variableDeclaration.node],
+  })
+  const updateAstResult = await dependencies.kclManager.updateAst(
+    modifiedAst,
+    false
+  )
+  await dependencies.codeManager.updateEditorWithAstAndWriteToFile(
+    updateAstResult.newAst
+  )
+  await dependencies.kclManager.updateAst(modifiedAst, true)
```

Comment:
Should be using @max-mrgrsk's updateModelingState function for these calls?

Reply:
done